js_modules/dagster-ui/packages/ui-core/src/schedules/ScheduleSwitch.tsx

```diff
@@ -22,18 +26,32 @@ interface Props {
   repoAddress: RepoAddress;
   schedule: ScheduleSwitchFragment;
   size?: 'small' | 'large';
+  hasOutdatedScheduleState?: boolean;
 }
 
 export const ScheduleSwitch = (props: Props) => {
-  const {repoAddress, schedule, size = 'large'} = props;
+  const {repoAddress, schedule, size = 'large', hasOutdatedScheduleState} = props;
   const {name, scheduleState} = schedule;
-  const {status, id, selectorId} = scheduleState;
+  const {id, selectorId} = scheduleState;
 
   const {
     permissions: {canStartSchedule, canStopRunningSchedule},
     disabledReasons,
   } = usePermissionsForLocation(repoAddress.location);
 
+  const scheduleSelector = {
+    ...repoAddressToSelector(repoAddress),
+    scheduleName: name,
+  };
+
+  const {data, loading} = useQuery<ScheduleStateQuery, ScheduleStateQueryVariables>(
+    SCHEDULE_STATE_QUERY,
+    {
+      variables: {scheduleSelector},
+      skip: !hasOutdatedScheduleState,
```

Comment:
This ScheduleSwitch component is used by places that don't supply outdated data so I added this prop to make the component fetch up to date data only in the left-nav case.

Reply:
`hasOutdatedScheduleState` -> `shouldFetchLatestState` / `refreshState` / `queryState` or something

"has outdated" miscommunicates that we know its outdated versus opting in to a behavior that resolves the potential problem of being outdated